.github/workflows/sphinx_docs.yml

```
@@ -0,0 +1,57 @@
+name: sphinx docs
+
+on:
+  push: # run on push to main and PR
+    branches:
+      - main
+    pull_request:
+
+env:
+  # Python version used to build sphinx docs
+  DOCS_PYTHON_VERSION: "3.12"
+
+jobs:
+  docs:
+    name: Sphinx documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.DOCS_PYTHON_VERSION }}
+      
+      # base the Python cache on the hash of all pyproject.toml,
+      # which includes Python requirements.
+      # if any change, the cache is invalidated.
+      - name: Cache pip
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('pyproject.toml') }}
+          restore-keys: |
+            pip-${{ hashFiles('pyproject.toml') }}
+            pip-
+
+      # TODO: Is installing the dev environment the right call?
+      # Install dependencies
+      - name: Install dependencies
+        run: pip install -e ".[dev]"
+
+      - name: Build Sphinx docs
+        run: cd docs && make -b coverage html
+
+      # For pull requests, exit with error if documentation coverage is incomplete
+      - name: Report on documentation coverage
+        if: ${{ github.event_name == 'pull_request' }}
+        run: if [[ $((`grep ppa _build/coverage/python.txt | grep -v 100 | wc -l`)) -eq 0 ]] ; then echo "Documentation coverage complete"; else cat sphinx-docs/_build/coverage/python.txt && exit 1; fi
```

Should the `ppa` here be changed to `corppa` ? The paths might need some adjustment also. Please try running this command locally and see what needs to be changed.

---

Might be something like this, but I'm guessing

```suggestion
        run: if [[ $((`grep corpppa _build/coverage/python.txt | grep -v 100 | wc -l`)) -eq 0 ]] ; then echo "Documentation coverage complete"; else cat docs/_build/coverage/python.txt && exit 1; fi
```

---

Good point, I missed this reference. Thanks!